opts/envfile_test.go

```diff
@@ -0,0 +1,104 @@
+package opts
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+// ParseEnvFile with a non existent file
+func TestParseEnvFileNonExistentFile(t *testing.T) {
+	_, err := ParseEnvFile("I don't exist as a file")
+	if err == nil {
+		t.Fatalf("Expected a PathError, got nothing")
+	}
+	if _, ok := err.(*os.PathError); !ok {
+		t.Fatalf("Expected a PathError, got [%v]", err)
+	}
+}
+
+// ParseEnvFile with an invalid file
+func TestParseEnvFileInvalidFile(t *testing.T) {
+	f, err := ioutil.TempFile("", "tail-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer f.Close()
+	defer os.RemoveAll(f.Name())
+	testFile := []byte(`ENV1=value1
+ENV 2=value with spaces`)
+	if _, err := f.Write(testFile); err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = ParseEnvFile(f.Name())
+	if err == nil {
+		t.Fatalf("Expected a ErrBadEnvVariable, got nothing")
+	}
+	if _, ok := err.(ErrBadEnvVariable); !ok {
+		t.Fatalf("Expected a ErrBadEnvVariable, got [%v]", err)
+	}
+	expectedMessage := "poorly formatted environment: variable 'ENV 2' has white spaces"
+	if err.Error() != expectedMessage {
+		t.Fatalf("Expected [%v], got [%v]", expectedMessage, err.Error())
+	}
+
+}
+
+// ParseEnvFile with a random file, pass through
+func TestParseEnvFileRandomFile(t *testing.T) {
+	f, err := ioutil.TempFile("", "tail-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer f.Close()
+	defer os.RemoveAll(f.Name())
+	testFile := []byte(`first line
+another invalid line`)
+	if _, err := f.Write(testFile); err != nil {
+		t.Fatal(err)
+	}
+
+	envs, err := ParseEnvFile(f.Name())
+
+	if err != nil {
```

Comment:
There is a *certain* consistency in the fact that we are able to run ``docker run -e "something with space" busybox /bin/sh`` ; although the environment variables is not set in the container. I should probably validate this too (``ValidateEnv``).